### DISCUSSION TITLE
マルチプレイSDKサンプルのエラーについて

### DISCUSSION THREAD

**Discussion by vdyakuin** [pixiv/vroid-sdk-developers#33] [Q&A] [answered]
お世話になっております。

マルチプレイSDKのサンプルを実行した際、エラーになってしまうのでご相談に乗っていただきたいのですが、こちらに詳細を書き込んでも問題ないでしょうか？それとも他に窓口がございますか？
（マルチプレイSDKは一部の方にしか提供していないとのことでしたので）

**Comment by Haresaku** [comment, marked as answer]
マルチプレイSDKは開発中の要素が多く、またセキュリティ上の確認事項も多いため、問い合わせフォームから詳細を伺ったのち、NDAを結び、SDKの提供へと進めています。

場合によってはこの場でお答えできないこともあるかもしれませんが、できる限り、SDKに関する議論がストックされるよう、Github Discussionsで質問いただければと思っております。ひとまず、この場で質問いただけますと幸いです。

**Reply by vdyakuin** [reply]
ご回答ありがとうございます。それでは詳細を記載いたします。

●やりたい事
マルチプレイSDKのサンプルにて、マルチプレイ用ダウンロードライセンスを発行したい。

●発生する問題
発行しようとすると、"OAUTH_FORBIDDEN" エラーになる。

●流れ
1. アプリ内からVRoid Hubにログインする。
2. ダウンロードしたいモデルを選択する
3. 「ダウンロードライセンスの新規発行」を押下（CreateDownloadLicenseが実行される）
4. HubMultiplayApi.PostDownloadLicenseMultiplayにてonErrorコールバックが返り、エラーの内容が以下のようになります。

code  "OAUTH_FORBIDDEN"
details Count = 0
message  "OAuth forbidden"



関係ありそうな情報を以下に記載します。


◆バージョン
Unity 2019.4.13f1 Personal
VRoidSDK-v0.0.21


◆アプリケーション管理ページ
スコープ
	default: デフォルト　　　　チェックあり
	heart: ハートの編集　　　　チェックなし
	multiplay: マルチプレイ　　チェックあり

リダイレクトURI
	urn:ietf:wg:oauth:2.0:oob	


◆Unityプロジェクト内
---SDK Configuration
Application Id　　　　アプリケーション管理ページから貼り付け済み
Secret 　　　　　　　アプリケーション管理ページから貼り付け済み
Android Url Scheme　　空白
IOS Url Scheme　　　　空白

Scope
	Multiplay 　　チェックあり
	Heart 　　　　チェックなし
	Default 　　　チェックあり
	
---MultiplaySdkConfiguration.AppPassword
設定済み


◆その他
WindowsPC Standaloneアプリケーションです。
マルチプレイ用の開発者権限は2021年12月下旬ごろ頂いております。
シングルプレイ用SDKでは、モデルのダウンロード、表示まで問題なく動きます。

**Reply by Haresaku** [reply]
詳細の記載ありがとうございます。一点確認していただきたいのが、連携後にMultiplay拡張を入れた後に連携のし直しはしましたでしょうか。
再連携しないとデバイスで保持しているトークンが破棄されないのでmultiplayの機能が使えないのでまだ連携のし直しをしてない場合は再連携して試してもらえると

**Reply by vdyakuin** [reply]
再連携を行ったところ、ライセンスの発行ができました。
また、そのライセンスを使用してモデルのダウンロードもできました。
ありがとうございます。
